# Silva-based classification without filtering

##Taxonomy assignment using Silva <br>
Since I am analyzing environmental data, I used the Silva database (132.99). I used the sklearn classification algorithm. <br>
I used the same ```--i-reads``` as detailed in Chapter2 (De novo clustering). These are seqs clustered using a 97% cutoff, but no filtering was performed on them:
```{bash eval=FALSE}
qiime feature-classifier classify-sklearn 
	--i-classifier /projects/e30740/qiime2_classifiers/silva-132-99-nb-classifier.qza 
	--i-reads /projects/e30740/mselensky/FinalProject/rep-LABEseqs-dn-97.qza
	--o-classification /projects/e30740/mselensky/FinalProject/NoFiltering/taxonomy_unfiltered.qza
```
I was not able to visualize the resultant taxonomy.qza file as a table using metadata tabulate. I had to employ the following workaround, which allowed me to visualize my taxonomy information in tabular form:
```{bash eval=FALSE}
qiime metadata tabulate 
	--m-input-file /projects/e30740/mselensky/FinalProject/NoFiltering/taxonomy_unfiltered.qza
	--o-visualization /projects/e30740/mselensky/FinalProject/NoFiltering/taxonomy_unfiltered.qzv
qiime tools export 
	--input-path /projects/e30740/mselensky/FinalProject/NoFiltering/taxonomy_unfiltered.qza 
	--output-path --output-path /projects/e30740/mselensky/FinalProject/NoFiltering/taxonomy-with-spaces
qiime metadata tabulate 
	--m-input-file /projects/e30740/mselensky/FinalProject/NoFiltering/taxonomy-with-spaces/taxonomy.tsv 
	--o-visualization /projects/e30740/mselensky/FinalProject/NoFiltering/taxonomy-as-metadata_unfiltered.qzv
qiime tools export 
	--input-path /projects/e30740/mselensky/FinalProject/NoFiltering/taxonomy-as-metadata_unfiltered.qzv 
	--output-path /projects/e30740/mselensky/FinalProject/NoFiltering/taxonomy-as-metadata
qiime tools import 
	--type 'FeatureData[Taxonomy]' 
	--input-path /projects/e30740/mselensky/FinalProject/NoFiltering/taxonomy-as-metadata/metadata.tsv 
	--output-path /projects/e30740/mselensky/FinalProject/NoFiltering/taxonomy-without-spaces_unfiltered.qza

	qiime metadata tabulate 
		--m-input-file /projects/e30740/mselensky/FinalProject/NoFiltering/taxonomy-without-spaces_unfiltered.qza 
		--o-visualization /projects/e30740/mselensky/FinalProject/NoFiltering/taxonomy-without-spaces_unfiltered.qzv
```

##Taxa barplot <br>
This yielded a taxa barplot as an output (before rarefaction):
```{bash eval=FALSE}
qiime taxa barplot 
	--i-table /projects/e30740/mselensky/FinalProject/NoFiltering/LABEtable_unaltered-dn-97.qza
	--i-taxonomy /projects/e30740/mselensky/FinalProject/NoFiltering/taxonomy_unfiltered.qza
	--m-metadata-file /projects/e30740/mselensky/FinalProject/LABEmetadata_all_samples.txt
	--o-visualization /projects/e30740/mselensky/FinalProject/NoFiltering/LABE_taxa_barplot_unfiltered.qzv
```

##Phylogenetic tree
```{bash eval=FALSE}
qiime phylogeny align-to-tree-mafft-fasttree
  --i-sequences /projects/e30740/mselensky/FinalProject/NoFiltering/rep-LABEseqs_unaltered-dn-97.qza
  --o-alignment /projects/e30740/mselensky/FinalProject/NoFiltering/Phylogeny/LABEseqs_unaltered-dn-97_aligned
  --o-masked-alignment /projects/e30740/mselensky/FinalProject/NoFiltering/Phylogeny/LABEseqs_unaltered-dn-97_masked_aligned
  --o-tree /projects/e30740/mselensky/FinalProject/NoFiltering/Phylogeny/LABEtree_unfiltered
  --o-rooted-tree /projects/e30740/mselensky/FinalProject/NoFiltering/Phylogeny/LABEtree_rooted_unfiltered
  --output-dir /projects/e30740/mselensky/FinalProject/NoFiltering/Phylogeny
```

##Core diversity metrics <br>
First, I made an alpha rarefaction plot to determine sampling depth:

```{bash eval=FALSE}
qiime diversity alpha-rarefaction
	--i-table /projects/e30740/mselensky/FinalProject/NoFiltering/LABEtable_unaltered-dn-97.qza
	--p-max-depth 5000
	--p-min-depth 1
	--p-steps 10
	--p-iterations 10
	--o-visualization /projects/e30740/mselensky/FinalProject/NoFiltering/LABE_alpha_rarefied_table_unfiltered
```
When sampling depth = 2000, <br>
* Retained 58,000 (50.74%) sequences in 29 (96.67%) samples at the specifed sampling depth. <br>
<br>
Using the phylogenetic tree from step 8, I then measured diversity core metrics on my rarefied data with a sampling depth of 2000:
```{bash eval=FALSE}
qiime diversity core-metrics-phylogenetic
--i-phylogeny /projects/e30740/mselensky/FinalProject/NoFiltering/Phylogeny/LABEtree_rooted_unfiltered.qza
--i-table /projects/e30740/mselensky/FinalProject/NoFiltering/LABEtable_unaltered-dn-97.qza
--p-sampling-depth 2000
--m-metadata-file /projects/e30740/mselensky/FinalProject/LABEmetadata_all_samples.txt
--output-dir /projects/e30740/mselensky/FinalProject/NoFiltering/CoreMetricsResults
```
**Outputs give the following information:** <br>

ALPHA DIVERSITY CORE METRICS: <br>

*Shannon's diversity index* <br> 
* A quantitative measure of community richness <br>

*Observed OTUs* <br> 
* A qualitative measure of community richness <br>

*Faith's Phylogenetic Diversity* <br> 
* A qualitiative measure of community richness that incorporates phylogenetic relationships between the features <br>

*Evenness (or Pielou's Evenness)* <br>
* A measure of community evenness <br>

-----------

BETA DIVERSITY CORE METRICS: <br>

*Jaccard distance* <br>
* A qualitative measure of community dissimilarity <br>

*Bray-Curtis* <br>
* A quantitative measure of community dissimilarity <br>

*Unweighted UniFrac distance* <br>
* A qualitative measure of community dissimilarity that incorporates phylogenetic relationships between features <br>

*Weighted UniFrac distance* <br>
* A quantitative measure of community dissimilarity that incorporates phylogenetic relationships between features

##Alpha diversity significance <br>
As in Chapter 2, I chose Shannon for all downstream alpha diversity analyses. I did not include other indices in my workflow this time to save space. <br>
Alpha diversity group significance:
```{bash eval=FALSE}
#Shannon
qiime diversity alpha-group-significance
  --i-alpha-diversity /projects/e30740/mselensky/FinalProject/NoFiltering/CoreMetricsResults/shannon_vector.qza
	--m-metadata-file /projects/e30740/mselensky/FinalProject/LABEmetadata_all_samples.txt
	--o-visualization /projects/e30740/mselensky/FinalProject/NoFiltering/CoreMetricsResults/shannon-group-significance_silv_unfiltered.qzv
```

Alpha diversity correlation:
```{bash eval=FALSE}
qiime diversity alpha-correlation
	--i-alpha-diversity /projects/e30740/mselensky/FinalProject/NoFiltering/CoreMetricsResults/shannon_vector.qza
	--m-metadata-file /projects/e30740/mselensky/FinalProject/LABEmetadata_all_samples.txt
	--o-visualization /projects/e30740/mselensky/FinalProject/NoFiltering/CoreMetricsResults/shannon-alpha-correlation.qzv 
```

##NMDS plots in R <br>
```{R eval=TRUE, echo=FALSE, collapse=TRUE, message=FALSE, include=FALSE}
library(ggplot2)
library(vegan)
library(funrar)
library(plotly)

rm(list=ls())

getwd()

setwd("C:/Users/msele/OneDrive/Documents/Winter2019_Classes/ANTHRO_490/Final_Project/core_metrics_unfiltered")

L2 <- read.csv("level_2_unfiltered.csv", header = TRUE, row.names = 1, check.names = FALSE)
L2_metadata <- read.csv("LABEmetadata_all_samples.txt", check.names = FALSE, fileEncoding = "UTF-8-BOM")

L2_abun <- L2[,c(-43:-55)]

is.data.frame(L2_abun)
L2_transposed <- t(L2_abun)

# Convert L2 (an absolute abundance table) into relative abundance table
mat = as.matrix(L2_transposed)
head(mat)  # Has absolute abundances
rel_mat = make_relative(mat)
head(rel_mat)  # Relative abundances

# Make_relative causes rel_mat to be atomic, not recursive. downstream data.frame only works on recursive objects 
is.recursive(rel_mat)
is.atomic(rel_mat)
rel_mat_recursive <- as.data.frame(rel_mat)
is.recursive(rel_mat_recursive)

rel_mat_recursive_t <- t(rel_mat_recursive)
is.recursive(rel_mat_recursive_t)
rel_mat_recursive_t <- as.data.frame(rel_mat_recursive_t)

#
grouping_info <- data.frame(row.names = rownames(rel_mat_recursive_t), t(as.data.frame(strsplit(rownames(rel_mat_recursive_t),"\t"))))
head(grouping_info)

NMDS_L2 <- metaMDS(rel_mat_recursive_t, distance = "bray", k = 2, try = 20, trymax = 100)
stressplot(NMDS_L2, pch = 19, cex=0.75, l.col = "tomato", p.col = "skyblue")

info_L2 = data.frame(x=NMDS_L2$point[,2],y=NMDS_L2$point[,1],SampleID = as.factor(grouping_info[,1]))

# Merge metadata file with info_L5, which contains NMDS coordinates
L2_merged <- merge(info_L2, L2_metadata, by.x = "SampleID", by.y = "SampleID")

# Plot NMDS
p <- ggplot(L2_merged, aes(x, y, label="", color = Type, group = Type, text = paste("Cave: ", Cave))) +
  geom_point() +
  scale_shape_manual(values=c(1:17)) +
  geom_text() +
  ggtitle("NMDS of Phylum-Level Taxonomy using Silva without Filtering") +
  theme(panel.background = element_rect(fill = "white")) +
  stat_ellipse(geom = "path")

p <- ggplotly(p)

p

```
Stressplot for Silva-based, unfiltered, phylum-level NMDS:
```{R eval=TRUE, echo=FALSE, collapse=TRUE, message=FALSE}
# Stress plot
stressplot(NMDS_L2, pch = 19, cex=0.75, l.col = "tomato", p.col = "skyblue")
```
<br>
Phylum-level NMDS of cave vs surface soil:
```{R eval=TRUE, echo=FALSE, collapse=TRUE, message=FALSE}
p <- ggplot(L2_merged, aes(x, y, label="", color = Type, group = Type, text = paste("Cave: ", Cave))) +
  geom_point() +
  scale_shape_manual(values=c(1:17)) +
  geom_text() +
  ggtitle("NMDS of Phylum-Level Taxonomy using Silva without Filtering") +
  theme(panel.background = element_rect(fill = "white")) +
  stat_ellipse(geom = "path")

p <- ggplotly(p)

p
```
<br>
Family-level NMDS plot:
```{R eval=TRUE, echo=FALSE, collapse=TRUE, message=FALSE, include=FALSE}
# ANTHRO 490 final project - NMDS plots
# Silva without Filtering
# Winter 2019

library(ggplot2)
library(vegan)
library(funrar)
library(plotly)

rm(list=ls())

getwd()

setwd("C:/Users/msele/OneDrive/Documents/Winter2019_Classes/ANTHRO_490/Final_Project/core_metrics_unfiltered")

L5 <- read.csv("level_5_unfiltered.csv", header = TRUE, row.names = 1, check.names = FALSE)
L5_metadata <- read.csv("LABEmetadata_all_samples.txt", check.names = FALSE, fileEncoding = "UTF-8-BOM")

L5_abun <- L5[,c(-726:-738)]

is.data.frame(L5_abun)
L5_transposed <- t(L5_abun)

# Convert L2 (an absolute abundance table) into relative abundance table
mat = as.matrix(L5_transposed)
head(mat)  # Has absolute abundances
rel_mat = make_relative(mat)
head(rel_mat)  # Relative abundances

# Make_relative causes rel_mat to be atomic, not recursive. downstream data.frame only works on recursive objects 
is.recursive(rel_mat)
is.atomic(rel_mat)
rel_mat_recursive <- as.data.frame(rel_mat)
is.recursive(rel_mat_recursive)

rel_mat_recursive_t <- t(rel_mat_recursive)
is.recursive(rel_mat_recursive_t)
rel_mat_recursive_t <- as.data.frame(rel_mat_recursive_t)

#
grouping_info <- data.frame(row.names = rownames(rel_mat_recursive_t), t(as.data.frame(strsplit(rownames(rel_mat_recursive_t),"\t"))))
head(grouping_info)

NMDS_L5 <- metaMDS(rel_mat_recursive_t, distance = "bray", k = 2, try = 20, trymax = 100)
stressplot(NMDS_L5, pch = 19, cex=0.75, l.col = "tomato", p.col = "skyblue")

info_L5 = data.frame(x=NMDS_L5$point[,2],y=NMDS_L5$point[,1],SampleID = as.factor(grouping_info[,1]))

# Merge metadata file with info_L5, which contains NMDS coordinates
L5_merged <- merge(info_L5, L5_metadata, by.x = "SampleID", by.y = "SampleID")

# Plot NMDS
p <- ggplot(L5_merged, aes(x, y, label="", color = Type, group = Type, text = paste("Cave: ", Cave))) +
  geom_point() +
  scale_shape_manual(values=c(1:17)) +
  geom_text() +
  ggtitle("NMDS of Family-Level Taxonomy using Silva without Filtering") +
  stat_ellipse(geom = "path")

p <- ggplotly(p)

p
```

```{R eval=TRUE, echo=FALSE, collapse=TRUE, message=FALSE}
p <- ggplot(L5_merged, aes(x, y, label="", color = Type, group = Type, text = paste("Cave: ", Cave))) +
  geom_point() +
  scale_shape_manual(values=c(1:17)) +
  geom_text() +
  ggtitle("NMDS of Family-Level Taxonomy using Silva without Filtering") +
  theme(panel.background = element_rect(fill = "white")) +
  stat_ellipse(geom = "path")

p <- ggplotly(p)

p
```

<br>
NMDS plot with ellipses indicating 95% confidence level at which ellipses were drawn:
```{R eval=TRUE, echo=FALSE, collapse=TRUE, message=FALSE}
# Plot NMDS
p <- ggplot(L5_merged, aes(x, y, label="", color = Class, group = Mat_color, text = paste("Cave: ", Cave))) +
  geom_point() +
 # scale_shape_manual(values=c(1:17)) +
  scale_color_manual(values=c("darkgoldenrod4", "tan", "grey88", "gold")) +
  geom_text() +
  ggtitle("Family-level NMDS using Silva (without Filtering)") +
  theme(panel.background = element_rect("white")) +
  stat_ellipse(geom = "path")

p <- ggplotly(p)

p
```
