# greengenes-based classification without filtering

##Taxonomy assignment using greengenes <br>
I am now using the greengenes database to "mimic" the approach used by Lavoie et al (2017). I used the sklearn classification algorithm. <br>
I used the same ```--i-reads``` as detailed in Chapter2 (De novo clustering). These are seqs clustered using a 97% cutoff, but no filtering was performed on them:
```{bash eval=FALSE}
qiime feature-classifier classify-sklearn 
	--i-classifier /projects/e30740/qiime2_classifiers/gg-13-8-99-ng-classifier.qza 
	--i-reads /projects/e30740/mselensky/FinalProject/rep-LABEseqs.qza
	--o-classification /projects/e30740/mselensky/FinalProject/NoFiltering/taxonomy_gg/taxonomy_gg.qza
```

##Taxa barplot <br>
This yielded a taxa barplot as an output (before rarefaction):
```{bash eval=FALSE}
qiime taxa barplot 
	--i-table /projects/e30740/mselensky/FinalProject/NoFiltering/LABEtable_unaltered-dn-97.qza
	--i-taxonomy /projects/e30740/mselensky/FinalProject/NoFiltering/taxonomy_gg/taxonomy_gg.qza
	--m-metadata-file /projects/e30740/mselensky/FinalProject/LABEmetadata_all_samples.txt
	--o-visualization /projects/e30740/mselensky/FinalProject/NoFiltering/greengenes/LABE_taxa_barplot_unfiltered.qzv
```

##Phylogenetic tree
```{bash eval=FALSE}
qiime phylogeny align-to-tree-mafft-fasttree
  --i-sequences /projects/e30740/mselensky/FinalProject/NoFiltering/rep-LABEseqs_unaltered-dn-97.qza
  --o-alignment /projects/e30740/mselensky/FinalProject/NoFiltering/greengenes/Phylogeny/LABEseqs_unaltered-dn-97_aligned_gg
  --o-masked-alignment /projects/e30740/mselensky/FinalProject/NoFiltering/greengenes/Phylogeny/LABEseqs_unaltered-dn-97_masked_aligned_gg
  --o-tree /projects/e30740/mselensky/FinalProject/NoFiltering/greengenes/Phylogeny/LABEtree_unfiltered_gg
  --o-rooted-tree /projects/e30740/mselensky/FinalProject/NoFiltering/greengenes/Phylogeny/LABEtree_rooted_unfiltered_gg
  --output-dir /projects/e30740/mselensky/FinalProject/NoFiltering/greengenes/Phylogeny
```

##Core diversity metrics <br>

Using the phylogenetic tree, I then measured diversity core metrics on my rarefied data with a sampling depth of 2000. I used the same sampling depth as before because my ```--i-table``` did not change.
```{bash eval=FALSE}
qiime diversity core-metrics-phylogenetic
	--i-phylogeny /projects/e30740/mselensky/FinalProject/NoFiltering/greengenes/Phylogeny/LABEtree_rooted_unfiltered_gg
	--i-table /projects/e30740/mselensky/FinalProject/NoFiltering/LABEtable_unaltered-dn-97.qza
	--p-sampling-depth 2000
	--m-metadata-file /projects/e30740/mselensky/FinalProject/LABEmetadata_all_samples.txt
	--output-dir /projects/e30740/mselensky/FinalProject/NoFiltering/greengenes/CoreMetricsResults
```
**Outputs give the following information:** <br>

ALPHA DIVERSITY CORE METRICS: <br>

*Shannon's diversity index* <br> 
* A quantitative measure of community richness <br>

*Observed OTUs* <br> 
* A qualitative measure of community richness <br>

*Faith's Phylogenetic Diversity* <br> 
* A qualitiative measure of community richness that incorporates phylogenetic relationships between the features <br>

*Evenness (or Pielou's Evenness)* <br>
* A measure of community evenness <br>

-----------

BETA DIVERSITY CORE METRICS: <br>

*Jaccard distance* <br>
* A qualitative measure of community dissimilarity <br>

*Bray-Curtis* <br>
* A quantitative measure of community dissimilarity <br>

*Unweighted UniFrac distance* <br>
* A qualitative measure of community dissimilarity that incorporates phylogenetic relationships between features <br>

*Weighted UniFrac distance* <br>
* A quantitative measure of community dissimilarity that incorporates phylogenetic relationships between features

##Alpha diversity significance <br>
As in Chapter 2, I chose Shannon for all downstream alpha diversity analyses. I did not include other indices in my workflow this time to save space. <br>
Alpha diversity group significance:
```{bash eval=FALSE}
#Shannon
qiime diversity alpha-group-significance
    --i-alpha-diversity /projects/e30740/mselensky/FinalProject/NoFiltering/greengenes/CoreMetricsResults/shannon_vector.qza
	--m-metadata-file /projects/e30740/mselensky/FinalProject/LABEmetadata_all_samples.txt
	--o-visualization /projects/e30740/mselensky/FinalProject/NoFiltering/greengenes/CoreMetricsResults/shannon-group-significance_gg_unfiltered.qzv
```

Alpha diversity correlation:
```{bash eval=FALSE}
qiime diversity alpha-correlation
	--i-alpha-diversity /projects/e30740/mselensky/FinalProject/NoFiltering/CoreMetricsResults/shannon_vector.qza
	--m-metadata-file /projects/e30740/mselensky/FinalProject/LABEmetadata_all_samples.txt
	--o-visualization /projects/e30740/mselensky/FinalProject/NoFiltering/CoreMetricsResults/shannon-alpha-correlation.qzv 
```

##Beta diversity significance 

##Importing data for R-based analyses <br>
I got a little sick of using Qiime on Quest, so I decided to do all further analyses in R. <br>

I needed to export my OTU table as a .tsv for downstream R analyses:
```{bash eval=FALSE}
qiime tools export 
  --input-path /projects/e30740/mselensky/FinalProject/uchime-dn-out/LABEtable-nonchimeric-wo-borderline.qza
  --output-path /projects/e30740/mselensky/FinalProject/Filtered

qiime tools export 
  --input-path /projects/e30740/mselensky/FinalProject/taxonomy/taxonomy.qza
  --output-path /projects/e30740/mselensky/FinalProject/taxonomy

biom add-metadata 
  -i /projects/e30740/mselensky/FinalProject/Filtered/feature-table.biom -o /projects/e30740/mselensky/FinalProject/Filtered/feature-table_with_taxa.biom 
  --observation-metadata-fp /projects/e30740/mselensky/FinalProject/taxonomy/taxonomy.tsv
  --sc-separated taxonomy --observation-header OTUID,taxonomy

biom convert 
  -i /projects/e30740/mselensky/FinalProject/Filtered/feature-table_with_taxa.biom 
  -o /projects/e30740/mselensky/FinalProject/Filtered/feature-table_with_taxa.txt 
  --to-tsv --header-key taxonomy
```
In python, I ran:
```{p eval=FALSE}
summarize_taxa.py 
  -i feature-table_with_taxa.biom 
  -a 
  -o feature-table_with_taxa_abs_summary/ -L 1,2,3,4,5,6,7   
  --suppress_biom_table_output
```
This provided me with 7 ```feature-table_with_taxa_L(x).txt``` files, where (x) = taxonomic level of classification. After doing this, I realized that view.qiime2.org allows you to directly convert .qza files to .csv...oops!

##NMDS plots in R - family-level taxonomy <br>
```{R eval=TRUE, echo=FALSE, collapse=TRUE, message=FALSE, include=FALSE}
library(ggplot2)
library(vegan)
library(funrar)
library(plotly)
#package_version('plotly')

rm(list=ls())

getwd()

setwd("C:/Users/msele/OneDrive/Documents/Winter2019_Classes/ANTHRO_490/Final_Project/taxonomy")

L5 <- read.delim("feature-table_with_taxa_L5.txt", sep="\t", header = TRUE, row.names = 1, check.names = FALSE)
L5_metadata <- read.csv("LABEmetadata_all_samples.txt", check.names = FALSE, fileEncoding = "UTF-8-BOM")

is.data.frame(L5)
L5_transposed <- t(L5)

# Convert L5 (an absolute abundance table) into relative abundance table
mat = as.matrix(L5_transposed)
rel_mat = make_relative(mat)

# Make_relative causes rel_mat to be atomic, not recursive. downstream data.frame only works on recursive objects 
is.recursive(rel_mat)
is.atomic(rel_mat)
rel_mat_recursive <- as.data.frame(rel_mat)
is.recursive(rel_mat_recursive)

grouping_info <- data.frame(row.names = rownames(rel_mat), t(as.data.frame(strsplit(rownames(rel_mat),"\t"))))
head(grouping_info)

NMDS_L5 <- metaMDS(rel_mat_recursive, distance = "bray", k = 2, try = 20, trymax = 100)
stressplot(NMDS_L5, pch = 19, cex=0.75, l.col = "tomato", p.col = "skyblue")

info_L5 = data.frame(x=NMDS_L5$point[,2],y=NMDS_L5$point[,1],SampleID = as.factor(grouping_info[,1]))

# Merge metadata file with info_L5, which contains NMDS coordinates
L5_merged <- merge(info_L5, L5_metadata, by.x = "SampleID", by.y = "SampleID")

```
Stressplot for Silva-based, filtered, family-level NMDS:
```{R eval=TRUE, echo=FALSE, collapse=TRUE, message=FALSE}
# Stress plot
stressplot(NMDS_L5, pch = 19, cex=0.75, l.col = "tomato", p.col = "skyblue")
```
<br>
NMDS of cave vs surface soil:
```{R eval=TRUE, echo=FALSE, collapse=TRUE, message=FALSE}
# Plot NMDS
p <- ggplot(L5_merged, aes(x, y, label="", color = Type, group = Type, text = paste("Cave: ", Cave))) +
  geom_point() +
 # scale_shape_manual(values=c(1:17)) +
  scale_color_manual(values=c("skyblue", "tomato", "skyblue", "tomato")) +
  geom_text() +
  ggtitle("Family-level NMDS using Silva (without Filtering)") +
  theme(panel.background = element_rect("white")) +
  stat_ellipse(geom = "path")



p <- ggplotly(p)


p
```
<br>
Family-level NMDS plot:
```{R eval=TRUE, echo=FALSE, collapse=TRUE, message=FALSE}
# Plot NMDS
p <- ggplot(L5_merged, aes(x, y, label="", color = Class, group = Mat_color, text = paste("Cave: ", Cave))) +
  geom_point() +
 # scale_shape_manual(values=c(1:17)) +
  scale_color_manual(values=c("darkgoldenrod4", "tan", "grey88", "gold")) +
  geom_text() +
  ggtitle("Family-level NMDS using Silva (without Filtering)") +
  theme(panel.background = element_rect("white"))
 # stat_ellipse(geom = "path")



p <- ggplotly(p)


p
```
<br>
NMDS plot with ellipses indicating 95% confidence level at which ellipses were drawn:
```{R eval=TRUE, echo=FALSE, collapse=TRUE, message=FALSE}
# Plot NMDS
p <- ggplot(L5_merged, aes(x, y, label="", color = Class, group = Mat_color, text = paste("Cave: ", Cave))) +
  geom_point() +
 # scale_shape_manual(values=c(1:17)) +
  scale_color_manual(values=c("darkgoldenrod4", "tan", "grey88", "gold")) +
  geom_text() +
  ggtitle("Family-level NMDS using Silva (without Filtering)") +
  theme(panel.background = element_rect("white")) +
  stat_ellipse(geom = "path")

p <- ggplotly(p)

p
```
