<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>Chapter 2 Silva-based classification with filtering | ANTHRO 490 Final Project</title>
  <meta name="description" content="This book details the workflow I followed in my final project for ANTHRO 490.">
  <meta name="generator" content="bookdown  and GitBook 2.6.7">

  <meta property="og:title" content="Chapter 2 Silva-based classification with filtering | ANTHRO 490 Final Project" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This book details the workflow I followed in my final project for ANTHRO 490." />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 2 Silva-based classification with filtering | ANTHRO 490 Final Project" />
  
  <meta name="twitter:description" content="This book details the workflow I followed in my final project for ANTHRO 490." />
  

<meta name="author" content="Matt Selensky">


<meta name="date" content="2019-03-19">

  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="index.html">
<link rel="next" href="silva-based-classification-without-filtering.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Minimal Book Example</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Project context and overview</a></li>
<li class="chapter" data-level="2" data-path="silva-based-classification-with-filtering.html"><a href="silva-based-classification-with-filtering.html"><i class="fa fa-check"></i><b>2</b> Silva-based classification with filtering</a><ul>
<li class="chapter" data-level="2.1" data-path="silva-based-classification-with-filtering.html"><a href="silva-based-classification-with-filtering.html#import-data"><i class="fa fa-check"></i><b>2.1</b> Import data: <br></a></li>
<li class="chapter" data-level="2.2" data-path="silva-based-classification-with-filtering.html"><a href="silva-based-classification-with-filtering.html#dereplicate-sequences"><i class="fa fa-check"></i><b>2.2</b> Dereplicate sequences:</a></li>
<li class="chapter" data-level="2.3" data-path="silva-based-classification-with-filtering.html"><a href="silva-based-classification-with-filtering.html#de-novo-clustering"><i class="fa fa-check"></i><b>2.3</b> De novo clustering:</a></li>
<li class="chapter" data-level="2.4" data-path="silva-based-classification-with-filtering.html"><a href="silva-based-classification-with-filtering.html#chimera-checking-and-removal"><i class="fa fa-check"></i><b>2.4</b> Chimera checking and removal: <br></a></li>
<li class="chapter" data-level="2.5" data-path="silva-based-classification-with-filtering.html"><a href="silva-based-classification-with-filtering.html#abundance-filtering"><i class="fa fa-check"></i><b>2.5</b> Abundance filtering</a></li>
<li class="chapter" data-level="2.6" data-path="silva-based-classification-with-filtering.html"><a href="silva-based-classification-with-filtering.html#taxonomy-assignment-using-silva"><i class="fa fa-check"></i><b>2.6</b> Taxonomy assignment using Silva <br></a></li>
<li class="chapter" data-level="2.7" data-path="silva-based-classification-with-filtering.html"><a href="silva-based-classification-with-filtering.html#taxa-barplot"><i class="fa fa-check"></i><b>2.7</b> Taxa barplot <br></a></li>
<li class="chapter" data-level="2.8" data-path="silva-based-classification-with-filtering.html"><a href="silva-based-classification-with-filtering.html#phylogenetic-tree"><i class="fa fa-check"></i><b>2.8</b> Phylogenetic tree</a></li>
<li class="chapter" data-level="2.9" data-path="silva-based-classification-with-filtering.html"><a href="silva-based-classification-with-filtering.html#core-diversity-metrics"><i class="fa fa-check"></i><b>2.9</b> Core diversity metrics <br></a></li>
<li class="chapter" data-level="2.10" data-path="silva-based-classification-with-filtering.html"><a href="silva-based-classification-with-filtering.html#alpha-diversity-significance"><i class="fa fa-check"></i><b>2.10</b> Alpha diversity significance</a></li>
<li class="chapter" data-level="2.11" data-path="silva-based-classification-with-filtering.html"><a href="silva-based-classification-with-filtering.html#beta-diversity-significance"><i class="fa fa-check"></i><b>2.11</b> Beta diversity significance</a></li>
<li class="chapter" data-level="2.12" data-path="silva-based-classification-with-filtering.html"><a href="silva-based-classification-with-filtering.html#importing-data-for-r-based-analyses"><i class="fa fa-check"></i><b>2.12</b> Importing data for R-based analyses <br></a></li>
<li class="chapter" data-level="2.13" data-path="silva-based-classification-with-filtering.html"><a href="silva-based-classification-with-filtering.html#nmds-plots-in-r"><i class="fa fa-check"></i><b>2.13</b> NMDS plots in R <br></a><ul>
<li class="chapter" data-level="2.13.1" data-path="silva-based-classification-with-filtering.html"><a href="silva-based-classification-with-filtering.html#phylum-level-nmds"><i class="fa fa-check"></i><b>2.13.1</b> Phylum-level NMDS</a></li>
<li class="chapter" data-level="2.13.2" data-path="silva-based-classification-with-filtering.html"><a href="silva-based-classification-with-filtering.html#family-level-nmds"><i class="fa fa-check"></i><b>2.13.2</b> Family-level NMDS</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="silva-based-classification-without-filtering.html"><a href="silva-based-classification-without-filtering.html"><i class="fa fa-check"></i><b>3</b> Silva-based classification without filtering</a><ul>
<li class="chapter" data-level="3.0.1" data-path="silva-based-classification-without-filtering.html"><a href="silva-based-classification-without-filtering.html#family-level-nmds-scripts"><i class="fa fa-check"></i><b>3.0.1</b> Family-level NMDS scripts: <br></a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="results.html"><a href="results.html"><i class="fa fa-check"></i><b>4</b> Results</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">ANTHRO 490 Final Project</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="silva-based-classification-with-filtering" class="section level1">
<h1><span class="header-section-number">Chapter 2</span> Silva-based classification with filtering</h1>
<div id="import-data" class="section level2">
<h2><span class="header-section-number">2.1</span> Import data: <br></h2>
<p>I must convert the provided FASTA file into .qza format, which is readable by Qiime2.</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="ex">qiime</span> tools import 
    <span class="ex">--input-path</span> LABEseqs.fna 
    <span class="ex">--output-path</span> LABEseqs.qza 
    <span class="ex">--type</span> <span class="st">&#39;SampleData[Sequences]&#39;</span></code></pre></div>
</div>
<div id="dereplicate-sequences" class="section level2">
<h2><span class="header-section-number">2.2</span> Dereplicate sequences:</h2>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="ex">qiime</span> vsearch dereplicate-sequences 
    <span class="ex">--i-sequences</span> LABEseqs.qza 
    <span class="ex">--o-dereplicated-table</span> LABEtable.qza 
    <span class="ex">--o-dereplicated-sequences</span> rep-LABEseqs.qza</code></pre></div>
</div>
<div id="de-novo-clustering" class="section level2">
<h2><span class="header-section-number">2.3</span> De novo clustering:</h2>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="ex">qiime</span> vsearch cluster-features-de-novo 
    <span class="ex">--i-table</span> LABEtable.qza 
    <span class="ex">--i-sequences</span> rep-LABEseqs.qza 
    <span class="ex">--p-perc-identity</span> 0.97 
    <span class="ex">--o-clustered-table</span> LABEtable-dn-97.qza 
    <span class="ex">--o-clustered-sequences</span> rep-LABEseqs-dn-97.qza</code></pre></div>
<p>Since I am using 454 data, I used a cutoff of 97% to avoid PCR-induced error in my clustering algorithm. <br></p>
</div>
<div id="chimera-checking-and-removal" class="section level2">
<h2><span class="header-section-number">2.4</span> Chimera checking and removal: <br></h2>
<p>Check for chimeras:</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="ex">qiime</span> vsearch uchime-denovo 
    <span class="ex">--i-table</span> LABEtable-dn-97.qza 
    <span class="ex">--i-sequences</span> rep-LABEseqs-dn-97.qza 
    <span class="ex">--output-dir</span> uchime-dn-out</code></pre></div>
<p>Visualize the resultant chimera check output:</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="ex">qiime</span> metadata tabulate 
    <span class="ex">--m-input-file</span> uchime-dn-out/stats.qza 
    <span class="ex">--o-visualization</span> uchime-dn-out/stats.qzv</code></pre></div>
<p>Number of samples: 30 <br> Number of features: 103,356 (sequences) <br> Total frequency: 127,603 (reads) <br> <br></p>
<p>Filter out chimeras and borderline chimeras:</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="ex">qiime</span> feature-table filter-features 
    <span class="ex">--i-table</span> LABEtable-dn-97.qza 
    <span class="ex">--m-metadata-file</span> uchime-dn-out/nonchimeras.qza 
    <span class="ex">--o-filtered-table</span> uchime-dn-out/LABEtable-nonchimeric-wo-borderline.qza
    
<span class="ex">qiime</span> feature-table filter-seqs 
    <span class="ex">--i-data</span> rep-LABEseqs-dn-97.qza 
    <span class="ex">--m-metadata-file</span> uchime-dn-out/nonchimeras.qza 
    <span class="ex">--o-filtered-data</span> uchime-dn-out/LABErep-seqs-nonchimeric-wo-borderline.qza</code></pre></div>
<p>Visualize the resultant chimera-free output:</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="ex">qiime</span> feature-table summarize 
    <span class="ex">--i-table</span> uchime-dn-out/LABEtable-nonchimeric-wo-borderline.qza 
    <span class="ex">--o-visualization</span> uchime-dn-out/table-nonchimeric-wo-borderline.qzv</code></pre></div>
<p>Number of samples: 30 <br> Number of features: 13,389 (OTUs) <br> Total frequency: 114,300 (reads) <br> <em>NOTE - Lavoie et al. 2017 report 140,848 OTUs.</em> <br></p>
<p><br></p>
</div>
<div id="abundance-filtering" class="section level2">
<h2><span class="header-section-number">2.5</span> Abundance filtering</h2>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="ex">qiime</span> feature-table filter-samples 
    <span class="ex">--i-table</span> /projects/e30740/mselensky/FinalProject/uchime-dn-out/LABEtable-nonchimeric-wo-borderline.qza 
    <span class="ex">--p-min-frequency</span> 100 
    <span class="ex">--o-filtered-table</span> /projects/e30740/mselensky/FinalProject/uchime-dn-out/sample-frequency100-filtered-LABEtable-dn-97.qza

<span class="ex">qiime</span> feature-table filter-seqs 
    <span class="ex">--i-data</span> /projects/e30740/mselensky/FinalProject/uchime-dn-out/LABErep-seqs-nonchimeric-wo-borderline.qza 
    <span class="ex">--p-min-frequency</span> 100 
    <span class="ex">--o-filtered-data</span> uchime-dn-out/LABErep-seqs-nonchimeric-wo-borderline.qza</code></pre></div>
</div>
<div id="taxonomy-assignment-using-silva" class="section level2">
<h2><span class="header-section-number">2.6</span> Taxonomy assignment using Silva <br></h2>
<p>Since I am analyzing environmental data, I used the Silva database (132.99). I used the sklearn classification algorithm.</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="ex">qiime</span> feature-classifier classify-sklearn 
    <span class="ex">--i-classifier</span> /projects/e30740/qiime2_classifiers/silva-132-99-nb-classifier.qza 
    <span class="ex">--i-reads</span> /projects/e30740/mselensky/FinalProject/uchime-dn-out/LABErep-seqs-nonchimeric-wo-borderline.qza 
    <span class="ex">--o-classification</span> /projects/e30740/mselensky/taxonomy/taxonomy.qza</code></pre></div>
<p>I was not able to visualize the resultant taxonomy.qza file as a table using metadata tabulate. I had to employ the following workaround, which allowed me to visualize my taxonomy information in tabular form:</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="ex">qiime</span> tools export 
    <span class="ex">--input-path</span> /projects/e30740/mselensky/FinalProject/taxonomy/taxonomy.qza 
    <span class="ex">--output-path</span> /projects/e30740/mselensky/FinalProject/taxonomy/taxonomy-with-spaces
<span class="ex">qiime</span> metadata tabulate 
    <span class="ex">--m-input-file</span> /projects/e30740/mselensky/FinalProject/taxonomy/taxonomy-with-spaces/taxonomy.tsv 
    <span class="ex">--o-visualization</span> /projects/e30740/mselensky/FinalProject/taxonomy/taxonomy-as-metadata.qzv
<span class="ex">qiime</span> tools export 
    <span class="ex">--input-path</span> /projects/e30740/mselensky/FinalProject/taxonomy/taxonomy-as-metadata.qzv 
    <span class="ex">--output-path</span> /projects/e30740/mselensky/FinalProject/taxonomy/taxonomy-as-metadata
<span class="ex">qiime</span> tools import 
    <span class="ex">--type</span> <span class="st">&#39;FeatureData[Taxonomy]&#39;</span> 
    <span class="ex">--input-path</span> /projects/e30740/mselensky/FinalProject/taxonomy/taxonomy-as-metadata/metadata.tsv 
    <span class="ex">--output-path</span> /projects/e30740/mselensky/FinalProject/taxonomy/taxonomy-without-spaces.qza

<span class="ex">qiime</span> metadata tabulate 
    <span class="ex">--m-input-file</span> /projects/e30740/mselensky/FinalProject/taxonomy/taxonomy-without-spaces.qza 
    <span class="ex">--o-visualization</span> /projects/e30740/mselensky/FinalProject/taxonomy/taxonomy-without-spaces.qzv</code></pre></div>
</div>
<div id="taxa-barplot" class="section level2">
<h2><span class="header-section-number">2.7</span> Taxa barplot <br></h2>
<p>This yielded a taxa barplot as an output (before rarefaction):</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="ex">qiime</span> taxa barplot 
    <span class="ex">--i-table</span> /projects/e30740/mselensky/FinalProject/uchime-dn-out/LABEtable-nonchimeric-wo-borderline.qza
    <span class="ex">--i-taxonomy</span> /projects/e30740/mselensky/FinalProject/taxonomy/taxonomy.qza
    <span class="ex">--m-metadata-file</span> /projects/e30740/mselensky/FinalProject/LABEmetadata.txt
    <span class="ex">--o-visualization</span> /projects/e30740/mselensky/FinalProject/taxonomy/LABE_taxa_barplot.qzv</code></pre></div>
</div>
<div id="phylogenetic-tree" class="section level2">
<h2><span class="header-section-number">2.8</span> Phylogenetic tree</h2>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="ex">qiime</span> phylogeny align-to-tree-mafft-fasttree
    <span class="ex">--i-sequences</span> /projects/e30740/mselensky/FinalProject/uchime-dn-out/LABErep-seqs-nonchimeric-wo-borderline.qza
    <span class="ex">--o-alignment</span> /projects/e30740/mselensky/FinalProject/Phylogeny/LABErep-seqs-nonchimeric-wo-borderline_aligned
    <span class="ex">--o-masked-alignment</span> /projects/e30740/mselensky/FinalProject/Phylogeny/LABErep-seqs-nonchimeric-wo-borderline_masked_aligned
    <span class="ex">--o-tree</span> /projects/e30740/mselensky/FinalProject/Phylogeny/LABEtree
    <span class="ex">--o-rooted-tree</span> /projects/e30740/mselensky/FinalProject/Phylogeny/LABEtree_rooted
    <span class="ex">--output-dir</span> /projects/e30740/mselensky/FinalProject/Phylogeny</code></pre></div>
</div>
<div id="core-diversity-metrics" class="section level2">
<h2><span class="header-section-number">2.9</span> Core diversity metrics <br></h2>
<p>First, I made an alpha rarefaction plot to determine sampling depth:</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="ex">qiime</span> diversity alpha-rarefaction
    <span class="ex">--i-table</span> /projects/e30740/mselensky/FinalProject/uchime-dn-out/LABEtable-nonchimeric-wo-borderline.qza
    <span class="ex">--p-max-depth</span> 5000
    <span class="ex">--p-min-depth</span> 1
    <span class="ex">--p-steps</span> 10
    <span class="ex">--p-iterations</span> 10
    <span class="ex">--o-visualization</span> /projects/e30740/mselensky/FinalProject/LABE_alpha_rarefied_table</code></pre></div>
<p>When sampling depth = 2000, <br> * Retained 58,000 (50.74%) sequences in 29 (96.67%) samples at the specifed sampling depth. <br> <br> Using the phylogenetic tree from step 8, I then measured diversity core metrics on my rarefied data with a sampling depth of 2000:</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="ex">qiime</span> diversity core-metrics-phylogenetic
<span class="ex">--i-phylogeny</span> /projects/e30740/mselensky/FinalProject/Phylogeny/LABEtree_rooted.qza
<span class="ex">--i-table</span> /projects/e30740/mselensky/FinalProject/uchime-dn-out/LABEtable-nonchimeric-wo-borderline.qza
<span class="ex">--p-sampling-depth</span> 2000
<span class="ex">--m-metadata-file</span> /projects/e30740/mselensky/FinalProject/LABEmetadata_all_samples.txt
<span class="ex">--output-dir</span> /projects/e30740/mselensky/FinalProject/CoreMetricsResults</code></pre></div>
<p><strong>Outputs give the following information:</strong> <br></p>
<p>ALPHA DIVERSITY CORE METRICS: <br></p>
<p><em>Shannon’s diversity index</em> <br> * A quantitative measure of community richness <br></p>
<p><em>Observed OTUs</em> <br> * A qualitative measure of community richness <br></p>
<p><em>Faith’s Phylogenetic Diversity</em> <br> * A qualitiative measure of community richness that incorporates phylogenetic relationships between the features <br></p>
<p><em>Evenness (or Pielou’s Evenness)</em> <br> * A measure of community evenness <br></p>
<hr />
<p>BETA DIVERSITY CORE METRICS: <br></p>
<p><em>Jaccard distance</em> <br> * A qualitative measure of community dissimilarity <br></p>
<p><em>Bray-Curtis</em> <br> * A quantitative measure of community dissimilarity <br></p>
<p><em>Unweighted UniFrac distance</em> <br> * A qualitative measure of community dissimilarity that incorporates phylogenetic relationships between features <br></p>
<p><em>Weighted UniFrac distance</em> <br> * A quantitative measure of community dissimilarity that incorporates phylogenetic relationships between features</p>
</div>
<div id="alpha-diversity-significance" class="section level2">
<h2><span class="header-section-number">2.10</span> Alpha diversity significance</h2>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="co">#Faith&#39;s Phylogenetic Diversity</span>
<span class="ex">qiime</span> diversity alpha-group-significance
    <span class="ex">--i-alpha-diversity</span> /projects/e30740/mselensky/FinalProject/CoreMetricsResults/faith_pd_vector.qza
    <span class="ex">--m-metadata-file</span> /projects/e30740/mselensky/FinalProject/LABEmetadata_all_samples.txt
    <span class="ex">--o-visualization</span> /projects/e30740/mselensky/FinalProject/CoreMetricsResults/faith-pd-group-significance.qzv

<span class="co"># Shannon</span>
<span class="ex">qiime</span> diversity alpha-group-significance
    <span class="ex">--i-alpha-diversity</span> /projects/e30740/mselensky/FinalProject/CoreMetricsResults/shannon_vector.qza
    <span class="ex">--m-metadata-file</span> /projects/e30740/mselensky/FinalProject/LABEmetadata_all_samples.txt
    <span class="ex">--o-visualization</span> /projects/e30740/mselensky/FinalProject/CoreMetricsResults/shannon-group-significance.qzv

<span class="co"># Pielou&#39;s evenness</span>
<span class="ex">qiime</span> diversity alpha-group-significance
    <span class="ex">--i-alpha-diversity</span> /projects/e30740/mselensky/FinalProject/CoreMetricsResults/evenness_vector.qza
    <span class="ex">--m-metadata-file</span> /projects/e30740/mselensky/FinalProject/LABEmetadata_all_samples.txt
    <span class="ex">--o-visualization</span> /projects/e30740/mselensky/FinalProject/CoreMetricsResults/evenness-group-significance.qzv</code></pre></div>
<p>I will use Shannon for downstream processes. It had the lowest p-value for all variables. <br></p>
<p>Visualize alpha diversity correlation:</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="ex">qiime</span> diversity alpha-correlation
    <span class="ex">--i-alpha-diversity</span> /projects/e30740/mselensky/FinalProject/CoreMetricsResults/shannon_vector.qza
    <span class="ex">--m-metadata-file</span> /projects/e30740/mselensky/FinalProject/LABEmetadata_all_samples.txt
    <span class="ex">--o-visualization</span> /projects/e30740/mselensky/FinalProject/CoreMetricsResults/shannon-alpha-correlation.qzv </code></pre></div>
</div>
<div id="beta-diversity-significance" class="section level2">
<h2><span class="header-section-number">2.11</span> Beta diversity significance</h2>
<p>Visualizations for UniFrac (weighted &amp; unweighted), Jaccard, and Bray Curtis were provided by <code>qiime diversity core-metrics-phylogenetic</code>.<br></p>
<p>I am interested in comparing the signficance of “Class” among my samples. “Class” provides information whether a sample came from a yellow/white/tan microbial mat, or from the overlying surface soil of a cave. <br></p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="ex">qiime</span> diversity beta-group-significance
<span class="ex">--i-distance-matrix</span> /projects/e30740/mselensky/FinalProject/CoreMetricsResults/unweighted_unifrac_distance_matrix.qza
<span class="ex">--m-metadata-file</span> /projects/e30740/mselensky/FinalProject/LABEmetadata_all_samples.txt
<span class="ex">--m-metadata-column</span> Class
<span class="ex">--p-pairwise</span>
<span class="ex">--o-visualization</span> /projects/e30740/mselensky/FinalProject/CoreMetricsResults/unweighted-unifrac-Class-significance.qzv</code></pre></div>
<p>Make PCoA from distance matrix:</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="ex">qiime</span> diversity pcoa
<span class="ex">--i-distance-matrix</span> /projects/e30740/mselensky/FinalProject/CoreMetricsResults/bray_curtis_distance_matrix.qza
<span class="ex">--o-pcoa</span> /projects/e30740/mselensky/FinalProject/CoreMetricsResults/bray_curtis_pcoa
<span class="ex">singularity</span> exec -B /projects/e30740 -B /projects/e30740/mselensky/FinalProject /projects/e30740/qiime2-core2018-8.simg qiime diversity pcoa --i-distance-matrix /projects/e30740/mselensky/FinalProject/CoreMetricsResults/bray_curtis_distance_matrix.qza --o-pcoa /projects/e30740/mselensky/FinalProject/CoreMetricsResults/bray_curtis_pcoa</code></pre></div>
<p>Plot Emperor PCoA for Distance from cave entrance.<br></p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="ex">qiime</span> emperor plot 
  <span class="ex">--i-pcoa</span> /projects/e30740/mselensky/FinalProject/CoreMetricsResults/bray_curtis_pcoa.qza
  <span class="ex">--m-metadata-file</span> /projects/e30740/mselensky/FinalProject/LABEmetadata_no_soils.txt
  <span class="ex">--p-custom-axes</span> Distance_from_entrance
  <span class="ex">--o-visualization</span> /projects/e30740/mselensky/FinalProject/Bray-Curtis-emperor-Distance_from_entrance.qzv</code></pre></div>
</div>
<div id="importing-data-for-r-based-analyses" class="section level2">
<h2><span class="header-section-number">2.12</span> Importing data for R-based analyses <br></h2>
<p>I got a little sick of using Qiime on Quest, so I decided to do all further analyses in R. <br></p>
<p>I needed to export my OTU table as a .tsv for downstream R analyses:</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="ex">qiime</span> tools export 
  <span class="ex">--input-path</span> /projects/e30740/mselensky/FinalProject/uchime-dn-out/LABEtable-nonchimeric-wo-borderline.qza
  <span class="ex">--output-path</span> /projects/e30740/mselensky/FinalProject/Filtered

<span class="ex">qiime</span> tools export 
  <span class="ex">--input-path</span> /projects/e30740/mselensky/FinalProject/taxonomy/taxonomy.qza
  <span class="ex">--output-path</span> /projects/e30740/mselensky/FinalProject/taxonomy

<span class="ex">biom</span> add-metadata 
  <span class="ex">-i</span> /projects/e30740/mselensky/FinalProject/Filtered/feature-table.biom -o /projects/e30740/mselensky/FinalProject/Filtered/feature-table_with_taxa.biom 
  <span class="ex">--observation-metadata-fp</span> /projects/e30740/mselensky/FinalProject/taxonomy/taxonomy.tsv
  <span class="ex">--sc-separated</span> taxonomy --observation-header OTUID,taxonomy

<span class="ex">biom</span> convert 
  <span class="ex">-i</span> /projects/e30740/mselensky/FinalProject/Filtered/feature-table_with_taxa.biom 
  <span class="ex">-o</span> /projects/e30740/mselensky/FinalProject/Filtered/feature-table_with_taxa.txt 
  <span class="ex">--to-tsv</span> --header-key taxonomy</code></pre></div>
<p>In python, I ran:</p>
<div class="sourceCode"><pre class="sourceCode p"><code class="sourceCode pascal">summarize_taxa.py 
  -i feature-table_with_taxa.biom 
  -a 
  -o feature-table_with_taxa_abs_summary/ -L <span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">4</span>,<span class="dv">5</span>,<span class="dv">6</span>,<span class="dv">7</span>   
  --suppress_biom_table_output</code></pre></div>
<p>This provided me with 7 <code>feature-table_with_taxa_L(x).txt</code> files, where (x) = taxonomic level of classification. After doing this, I realized that view.qiime2.org allows you to directly convert .qza files to .csv based on taxonomic level…oops!</p>
</div>
<div id="nmds-plots-in-r" class="section level2">
<h2><span class="header-section-number">2.13</span> NMDS plots in R <br></h2>
<div id="phylum-level-nmds" class="section level3">
<h3><span class="header-section-number">2.13.1</span> Phylum-level NMDS</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># ANTHRO 490 final project - NMDS plots</span>
<span class="co"># Silva with Filtering</span>
<span class="co"># Phylum-level NMDS</span>
<span class="co"># Winter 2019</span>

<span class="kw">library</span>(ggplot2)
<span class="kw">library</span>(vegan)
<span class="kw">library</span>(funrar)
<span class="kw">library</span>(plotly)

<span class="kw">rm</span>(<span class="dt">list=</span><span class="kw">ls</span>())

<span class="kw">getwd</span>()

<span class="kw">setwd</span>(<span class="st">&quot;C:/Users/msele/OneDrive/Documents/Winter2019_Classes/ANTHRO_490/Final_Project/taxonomy&quot;</span>)

L2 &lt;-<span class="st"> </span><span class="kw">read.delim</span>(<span class="st">&quot;feature-table_with_taxa_L2.txt&quot;</span>, <span class="dt">sep=</span><span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>, <span class="dt">header =</span> <span class="ot">TRUE</span>, <span class="dt">row.names =</span> <span class="dv">1</span>, <span class="dt">check.names =</span> <span class="ot">FALSE</span>)
L2_metadata &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="st">&quot;LABEmetadata_all_samples.txt&quot;</span>, <span class="dt">check.names =</span> <span class="ot">FALSE</span>, <span class="dt">fileEncoding =</span> <span class="st">&quot;UTF-8-BOM&quot;</span>)

<span class="kw">is.data.frame</span>(L2)
L2_transposed &lt;-<span class="st"> </span><span class="kw">t</span>(L2)

<span class="co"># Convert L5 (an absolute abundance table) into relative abundance table</span>
mat =<span class="st"> </span><span class="kw">as.matrix</span>(L2_transposed)
<span class="kw">head</span>(mat)  <span class="co"># Has absolute abundances</span>
rel_mat =<span class="st"> </span><span class="kw">make_relative</span>(mat)
<span class="kw">head</span>(rel_mat)  <span class="co"># Relative abundances</span>

<span class="co"># Make_relative causes rel_mat to be atomic, not recursive. downstream data.frame only works on recursive objects </span>
<span class="kw">is.recursive</span>(rel_mat)
<span class="kw">is.atomic</span>(rel_mat)
rel_mat_recursive &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(rel_mat)
<span class="kw">is.recursive</span>(rel_mat_recursive)

<span class="co">#</span>
grouping_info &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">row.names =</span> <span class="kw">rownames</span>(rel_mat), <span class="kw">t</span>(<span class="kw">as.data.frame</span>(<span class="kw">strsplit</span>(<span class="kw">rownames</span>(rel_mat),<span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>))))
<span class="kw">head</span>(grouping_info)


NMDS_L2 &lt;-<span class="st"> </span><span class="kw">metaMDS</span>(rel_mat_recursive, <span class="dt">distance =</span> <span class="st">&quot;bray&quot;</span>, <span class="dt">k =</span> <span class="dv">2</span>, <span class="dt">try =</span> <span class="dv">20</span>, <span class="dt">trymax =</span> <span class="dv">100</span>)
<span class="kw">stressplot</span>(NMDS_L2, <span class="dt">pch =</span> <span class="dv">19</span>, <span class="dt">cex=</span><span class="fl">0.75</span>, <span class="dt">l.col =</span> <span class="st">&quot;tomato&quot;</span>, <span class="dt">p.col =</span> <span class="st">&quot;skyblue&quot;</span>)

info_L2 =<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">x=</span>NMDS_L2<span class="op">$</span>point[,<span class="dv">2</span>],<span class="dt">y=</span>NMDS_L2<span class="op">$</span>point[,<span class="dv">1</span>],<span class="dt">SampleID =</span> <span class="kw">as.factor</span>(grouping_info[,<span class="dv">1</span>]))

p &lt;-<span class="st"> </span><span class="kw">ggplot</span>(L2_merged, <span class="kw">aes</span>(x, y, <span class="dt">label=</span>SampleID, <span class="dt">color =</span> Type, <span class="dt">group =</span> Type, <span class="dt">text =</span> <span class="kw">paste</span>(<span class="st">&quot;Cave: &quot;</span>, Cave))) <span class="op">+</span>
<span class="st">  </span><span class="kw">geom_point</span>() <span class="op">+</span>
<span class="st">  </span><span class="kw">scale_shape_manual</span>(<span class="dt">values=</span><span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">17</span>)) <span class="op">+</span>
<span class="st">  </span><span class="co">#geom_text() +</span>
<span class="st">  </span><span class="kw">ggtitle</span>(<span class="st">&quot;NMDS of Phylum-Level Taxonomy using Silva with Filtering&quot;</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">theme</span>(<span class="dt">panel.background =</span> <span class="kw">element_rect</span>(<span class="st">&quot;white&quot;</span>)) <span class="op">+</span>
<span class="st">  </span><span class="kw">stat_ellipse</span>(<span class="dt">geom =</span> <span class="st">&quot;path&quot;</span>)

p &lt;-<span class="st"> </span><span class="kw">ggplotly</span>(p)

p</code></pre></div>
</div>
<div id="family-level-nmds" class="section level3">
<h3><span class="header-section-number">2.13.2</span> Family-level NMDS</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(ggplot2)
<span class="kw">library</span>(vegan)
<span class="kw">library</span>(funrar)
<span class="kw">library</span>(plotly)
<span class="co">#package_version(&#39;plotly&#39;)</span>

<span class="kw">rm</span>(<span class="dt">list=</span><span class="kw">ls</span>())

<span class="kw">getwd</span>()

<span class="kw">setwd</span>(<span class="st">&quot;C:/Users/msele/OneDrive/Documents/Winter2019_Classes/ANTHRO_490/Final_Project/taxonomy&quot;</span>)

L5 &lt;-<span class="st"> </span><span class="kw">read.delim</span>(<span class="st">&quot;feature-table_with_taxa_L5.txt&quot;</span>, <span class="dt">sep=</span><span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>, <span class="dt">header =</span> <span class="ot">TRUE</span>, <span class="dt">row.names =</span> <span class="dv">1</span>, <span class="dt">check.names =</span> <span class="ot">FALSE</span>)
L5_metadata &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="st">&quot;LABEmetadata_all_samples.txt&quot;</span>, <span class="dt">check.names =</span> <span class="ot">FALSE</span>, <span class="dt">fileEncoding =</span> <span class="st">&quot;UTF-8-BOM&quot;</span>)

<span class="kw">is.data.frame</span>(L5)
L5_transposed &lt;-<span class="st"> </span><span class="kw">t</span>(L5)

<span class="co"># Convert L5 (an absolute abundance table) into relative abundance table</span>
mat =<span class="st"> </span><span class="kw">as.matrix</span>(L5_transposed)
rel_mat =<span class="st"> </span><span class="kw">make_relative</span>(mat)

<span class="co"># Make_relative causes rel_mat to be atomic, not recursive. downstream data.frame only works on recursive objects </span>
<span class="kw">is.recursive</span>(rel_mat)
<span class="kw">is.atomic</span>(rel_mat)
rel_mat_recursive &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(rel_mat)
<span class="kw">is.recursive</span>(rel_mat_recursive)

grouping_info &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">row.names =</span> <span class="kw">rownames</span>(rel_mat), <span class="kw">t</span>(<span class="kw">as.data.frame</span>(<span class="kw">strsplit</span>(<span class="kw">rownames</span>(rel_mat),<span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>))))
<span class="kw">head</span>(grouping_info)

NMDS_L5 &lt;-<span class="st"> </span><span class="kw">metaMDS</span>(rel_mat_recursive, <span class="dt">distance =</span> <span class="st">&quot;bray&quot;</span>, <span class="dt">k =</span> <span class="dv">2</span>, <span class="dt">try =</span> <span class="dv">20</span>, <span class="dt">trymax =</span> <span class="dv">100</span>)
<span class="kw">stressplot</span>(NMDS_L5, <span class="dt">pch =</span> <span class="dv">19</span>, <span class="dt">cex=</span><span class="fl">0.75</span>, <span class="dt">l.col =</span> <span class="st">&quot;tomato&quot;</span>, <span class="dt">p.col =</span> <span class="st">&quot;skyblue&quot;</span>)

info_L5 =<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">x=</span>NMDS_L5<span class="op">$</span>point[,<span class="dv">2</span>],<span class="dt">y=</span>NMDS_L5<span class="op">$</span>point[,<span class="dv">1</span>],<span class="dt">SampleID =</span> <span class="kw">as.factor</span>(grouping_info[,<span class="dv">1</span>]))

<span class="co"># Merge metadata file with info_L5, which contains NMDS coordinates</span>
L5_merged &lt;-<span class="st"> </span><span class="kw">merge</span>(info_L5, L5_metadata, <span class="dt">by.x =</span> <span class="st">&quot;SampleID&quot;</span>, <span class="dt">by.y =</span> <span class="st">&quot;SampleID&quot;</span>)</code></pre></div>
<p>Stressplot for Silva-based, filtered, family-level NMDS:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Stress plot</span>
<span class="kw">stressplot</span>(NMDS_L5, <span class="dt">pch =</span> <span class="dv">19</span>, <span class="dt">cex=</span><span class="fl">0.75</span>, <span class="dt">l.col =</span> <span class="st">&quot;tomato&quot;</span>, <span class="dt">p.col =</span> <span class="st">&quot;skyblue&quot;</span>)</code></pre></div>
<p><br> NMDS of cave vs surface soil (family-level):</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Plot NMDS</span>
p &lt;-<span class="st"> </span><span class="kw">ggplot</span>(L5_merged, <span class="kw">aes</span>(x, y, <span class="dt">label=</span><span class="st">&quot;&quot;</span>, <span class="dt">color =</span> Type, <span class="dt">group =</span> Type, <span class="dt">text =</span> <span class="kw">paste</span>(<span class="st">&quot;Cave: &quot;</span>, Cave))) <span class="op">+</span>
<span class="st">  </span><span class="kw">geom_point</span>() <span class="op">+</span>
<span class="st"> </span><span class="co"># scale_shape_manual(values=c(1:17)) +</span>
<span class="st">  </span><span class="kw">scale_color_manual</span>(<span class="dt">values=</span><span class="kw">c</span>(<span class="st">&quot;skyblue&quot;</span>, <span class="st">&quot;tomato&quot;</span>, <span class="st">&quot;skyblue&quot;</span>, <span class="st">&quot;tomato&quot;</span>)) <span class="op">+</span>
<span class="st">  </span><span class="kw">geom_text</span>() <span class="op">+</span>
<span class="st">  </span><span class="kw">ggtitle</span>(<span class="st">&quot;Family-level NMDS using Silva (with Filtering)&quot;</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">theme</span>(<span class="dt">panel.background =</span> <span class="kw">element_rect</span>(<span class="st">&quot;white&quot;</span>)) <span class="op">+</span>
<span class="st">  </span><span class="kw">stat_ellipse</span>(<span class="dt">geom =</span> <span class="st">&quot;path&quot;</span>)



p &lt;-<span class="st"> </span><span class="kw">ggplotly</span>(p)


p</code></pre></div>
<p><br> NMDS plot with ellipses indicating 95% confidence level at which ellipses were drawn:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Plot NMDS</span>
p &lt;-<span class="st"> </span><span class="kw">ggplot</span>(L5_merged, <span class="kw">aes</span>(x, y, <span class="dt">label=</span><span class="st">&quot;&quot;</span>, <span class="dt">color =</span> Class, <span class="dt">group =</span> Mat_color, <span class="dt">text =</span> <span class="kw">paste</span>(<span class="st">&quot;Cave: &quot;</span>, Cave))) <span class="op">+</span>
<span class="st">  </span><span class="kw">geom_point</span>() <span class="op">+</span>
<span class="st"> </span><span class="co"># scale_shape_manual(values=c(1:17)) +</span>
<span class="st">  </span><span class="kw">scale_color_manual</span>(<span class="dt">values=</span><span class="kw">c</span>(<span class="st">&quot;darkgoldenrod4&quot;</span>, <span class="st">&quot;tan&quot;</span>, <span class="st">&quot;grey88&quot;</span>, <span class="st">&quot;gold&quot;</span>)) <span class="op">+</span>
<span class="st">  </span><span class="kw">geom_text</span>() <span class="op">+</span>
<span class="st">  </span><span class="kw">ggtitle</span>(<span class="st">&quot;Family-level NMDS using Silva (with Filtering)&quot;</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">theme</span>(<span class="dt">panel.background =</span> <span class="kw">element_rect</span>(<span class="st">&quot;white&quot;</span>)) <span class="op">+</span>
<span class="st">  </span><span class="kw">stat_ellipse</span>(<span class="dt">geom =</span> <span class="st">&quot;path&quot;</span>)

p &lt;-<span class="st"> </span><span class="kw">ggplotly</span>(p)

p</code></pre></div>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="index.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="silva-based-classification-without-filtering.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/01-intro.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"download": ["bookdown-demo.pdf", "bookdown-demo.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
